src/Lean/PrettyPrinter/Delaborator/Builtins.lean

```diff
@@ -1386,25 +1386,30 @@ where
     let e@(.forallE n d e' i) ← getExpr | unreachable!
     let n ← if bindingNames.contains n then withFreshMacroScope <| MonadQuotation.addMacroScope n else pure n
     let bindingNames := bindingNames.insert n
-    let stxN := mkIdent n
-    let curIds := curIds.push ⟨stxN⟩
     if shouldGroupWithNext bindingNames e e' then
-      withBindingBody n <| delabParamsAux bindingNames idStx groups curIds
+      withBindingBody' n (mkAnnotatedIdent n) <| fun stxN => delabParamsAux bindingNames idStx groups (curIds.push ⟨stxN⟩)
     else
-      let group ← withBindingDomain do
+      let mkGroup ← withBindingDomain do
```

Comment:
Reading this PR online, I feel that some kind of complexity threshold has been passed here. I think a type annotation on `mkGroup` would make this code much easier to read.